mdp/hinet/layer.py

```diff
@@ -49,6 +50,23 @@ def __init__(self, nodes, dtype=None):
             input_dim += node.input_dim
             self.node_input_dims[index] = node.input_dim
         output_dim = self._get_output_dim_from_nodes()
+
+        # store which nodes are pretrained up to what phase
+        _pretrained_phase = [node.get_current_train_phase()
+                             for node in nodes]
+        # check if all the nodes are already fully trained
+        train_len = 0
+        for i_node, node in enumerate(nodes):
+            if node.is_trainable():
+                train_len += (len(node._get_train_seq())
+                              - _pretrained_phase[i_node])
```

Comment:
You can inline the get_current_train_phase call here, since you don't really need the _pretrained_phase list. Or am I missing something?